transaction/core/Cargo.toml

```diff
@@ -43,23 +47,17 @@ bulletproofs-og = { version = "3.0.0-pre.1", default-features = false }
 [target.'cfg(any(target_feature = "avx2", target_feature = "avx"))'.dependencies]
 curve25519-dalek = { version = "4.0.0-pre.2", default-features = false, features = ["simd_backend", "nightly"] }
 
-[dev-dependencies.proptest]
-version = "1.0" # Only works for 0.9.1 or newer
-default-features = false
-# Enable all default features not known to break code coverage builds
-features = ["default-code-coverage"]
-
 [target.'cfg(not(any(target_feature = "avx2", target_feature = "avx")))'.dependencies]
 curve25519-dalek = { version = "4.0.0-pre.2", default-features = false, features = ["nightly", "u64_backend"] }
 
 [dev-dependencies]
 assert_matches = "1.5"
+proptest = { version = "1.0", default-features = false, features = ["default-code-coverage"] }
 rand = "0.8"
-rand_hc = "0.3"
 tempdir = "0.3"
 
 mc-crypto-digestible-test-utils = { path = "../../crypto/digestible/test-utils" }
-mc-crypto-rand = { path = "../../crypto/rand" }
```

Comment:
we were able to break dependency on `mc-crypto-rand` and `rand_hc` by using RNG from `mc_util_test_helper`. we weren't able to completely remove `rand` in this revision because the blockchain hash consistency tests are using `StdRng`, that is tracked in a ticket now